File: leetcode/2260-minimum-consecutive-cards-to-pick-up/solution.ts
function minimumCardPickup(cards: number[]): number {
  const m = new Map<number, number>();

  let ans = Infinity;

  for (let i = 0; i < cards.length; i++) {
    if (m.has(cards[i])) {
      ans = Math.min(ans, i - m.get(cards[i]) + 1);
    }

    m.set(cards[i], i);
  }

  return ans == Infinity ? -1 : ans;
}

File: leetcode/2260-minimum-consecutive-cards-to-pick-up/solution.py
def minimumCardPickup(self, cards: List[int]) -> int:
    m = {}
    NO_ANS = 10000009
    ans = NO_ANS

    for i in range(len(cards)):
        if cards[i] in m:
            ans = min(ans, i - m[cards[i]] + 1)
        m[cards[i]] = i

    return -1 if ans == NO_ANS else ans
